MetaMask/metamask-extension
# Adding sizes to `ModalContent` component

## **Description**
Currently the `ModalContent` component only has one size: sm. This PR updates the `ModalContent` component to include different sizes: sm, md, lg
- Adds sizes: sm `360px` , md `480px`, lg `720px` 
- Also included recently merged `ModalBody` in `ModalContent` and `Modal` code examples

## **Related issues**

Part of: #19413 
[Figma design](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?type=design&node-id=15111%3A38404&mode=design&t=QgAvTsEhrx0zcIoo-1)

## **Manual testing steps**

1. Go to the  latest storybook build in this PR or pull this branch  <img width="953" alt="Screenshot 2023-12-06 at 4 08 58 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/77ce4927-d62a-4584-8aca-5c1ffe6ce1b5">
2. Search for `ModalContent` in the search bar (make sure it's the component-library version)
3. Check `size` docs, controls and stories

## **Screenshots/Recordings**

### **Before**

Before the `ModalContent` only had once size: sm as there were no other use cases.

https://github.com/MetaMask/metamask-extension/assets/8112138/345ad85a-00de-4844-aade-3613bcd1815f


### **After**


https://github.com/MetaMask/metamask-extension/assets/8112138/a30f0333-9547-4fd0-84b9-87f84c31848f

Testing coverage 
Will also improve once we move the `onClickOutsideLogic` to `ModalOverlay`. Size prop is covered 
<img width="757" alt="Screenshot 2023-12-06 at 3 47 50 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/7c0022c5-34bd-4826-a921-fb94af1e784d">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
